### ISSUE TITLE
Steps do not work for Windows 11 after update 22H2

### ISSUE THREAD

**Issue by CoolDadTx** [MicrosoftDocs/windows-itpro-docs#10988]
Somewhere around 22H2 Device Guard was changed such that attempting to use remote desktop with a saved credential no longer works. All the online docs say to disable Device Guard and eventually link to this page. However following the steps given here to disable the policy via Group Policy doesn't actually work. After the reboot the problem persists.

Given that all online resources seem to be referencing Device Guard disabling the docs should be updated to either ensure they work for 22H2+ or to have explicit steps to allow saved credentials to work post 22H2 so others don't waste time following these steps just to find out they don't actually work.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 41c2cb44-269d-3334-019a-726eb5432f1c
* Version Independent ID: 251024cc-b818-18c6-a112-4ac0276cb4b1
* Content: [Manage Windows Defender Credential Guard (Windows) - Windows security](https://learn.microsoft.com/en-us/windows/security/identity-protection/credential-guard/credential-guard-manage)
* Content Source: [windows/security/identity-protection/credential-guard/credential-guard-manage.md](https://github.com/MicrosoftDocs/windows-itpro-docs/blob/public/windows/security/identity-protection/credential-guard/credential-guard-manage.md)
* Product: **windows-client**
* GitHub Login: @paolomatarazzo
* Microsoft Alias: **paoloma**

**Comment by paolomatarazzo** [comment]
Thank you for the feedback @CoolDadTx . I'll follow up with the feature owner and update the article accordingly.

**Comment by paolomatarazzo** [comment]
@CoolDadTx did you check the entire article, including the UEFI lock piece? https://learn.microsoft.com/en-us/windows/security/identity-protection/credential-guard/credential-guard-manage#disabling-windows-defender-credential-guard-with-uefi-lock
Can you please confirm if you have UEFI lock enabled?

**Comment by CoolDadTx** [comment]
@paolomatarazzo AFAIK I am not using UEFI lock? 

![image](https://user-images.githubusercontent.com/3778674/200944794-b5dba895-303d-4689-a708-91c4b3119336.png)

I also checked the 2 mentioned registry entries and both values are already 0.

**Comment by paolomatarazzo** [comment]
**If** the device you are testing from has have UEFI lock, you must also follow step#3: "Delete the Windows Defender Credential Guard EFI variables by using bcdedit. From an elevated command prompt, type the following commands:". Did you do that, by any chance?

**Comment by CoolDadTx** [comment]
@paolomatarazzo AFAIK this machine has never used any of that. It was literally working on Win 11 until the latest update and then suddenly remote desktop doesn't like saved credentials anymore. So how can I tell if this was every enabled before without having to run a command that modifies my system to remove something that probably wasn't there to begin with?

**Comment by paolomatarazzo** [comment]
@CoolDadTx I'm checking with the devs

**Comment by CoolDadTx** [comment]
@RAJU2529 this machine isn't on a domain and therefore the change was applied to the local GP. I've never heard the need for forcing a GP update for a local change. Irrelevant the machine has been rebooted multiple times since the change was made.

**Comment by RAJU2529** [comment]
@CoolDadTx . That applying force parameter is necessary even in standardalone computers

**Comment by CoolDadTx** [comment]
@RAJU2529 Please provide the documentation that states this. My post here was to get the docs updated related to Win11 and if this is needed in the docs then that is sort of the point of this report. As I mentioned, I have been changing local policies for years on a variety of system issues and htis is the first time I've ever heard a local policy change not taking effect after reboot without using a gpupdate which would normally overwrite the local policy with a domain policy (which is already overridden by the local poilcy). Irrelevant I cannot test this until next week either way.

**Comment by RAJU2529** [comment]
@CoolDadTx 

**https://learn.microsoft.com/en-us/windows/security/threat-protection/device-guard/enable-virtualization-based-protection-of-code-integrity**

**To apply the new policy on a domain-joined computer, either restart or run gpupdate /force in an elevated command prompt.**

**Comment by CoolDadTx** [comment]
@RAJU2529 as I mentioned this is Not a domain joined machine and I have rebooted many times since then. Is the docs wrong?

**Comment by RAJU2529** [comment]
@CoolDadTx . I cannot comment on docs. But I  am using standard alone computer, use gpupdate force parameter, after modifying group policy

**Comment by paolomatarazzo** [comment]
@CoolDadTx if you already rebooted, then the local GPO has already applied. I don't have any indications that the article is wrong as I don't have any evidence of others reporting issues after we updated the doc. Since you are not sure if your machine has UEFI lock and, understandably, don't want to make the suggested changes unless needed, I'd ask you to wait until I hear back from the devs if there is a command to determine the status of UEFI lock.

**Comment by paolomatarazzo** [comment]
@CoolDadTx to check if Credential Guard has been enabled with UEFI Lock, you can look in the System log:
Source: Wininit
Event ID: 14

Which value do you see?

**Comment by CoolDadTx** [comment]
@paolomatarazzo 

Registry Configuration 0x2
Test Configuration 0
Auto Enablement 1

**Comment by paolomatarazzo** [comment]
Thank you @CoolDadTx, 0x2 is Enabled without UEFI Lock. I'll ask our developers to take a closer look at your case and to chime into this discussion. Thank you.

**Comment by paolomatarazzo** [comment]
@CoolDadTx : in the meantime, can you please share the following screenshots:
Even viewer of the event above
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Lsa\LsaCfgFlags
HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\DeviceGuard\LsaCfgFlags

**Comment by jirenugo** [comment]
Your device most likely does not explicitly disable Credential Guard. Autoenablement tells us that it was auto-enabled and the configuration value tells us that it was enabled without UEFI lock. As paolo has indicated, sharing those reg keys will be the next step. My guess is that you don't have LsaCfgFlags under HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Lsa. Setting that value to 0 will disable credential guard explicitly.

**Comment by CoolDadTx** [comment]
Screenshots

![image](https://user-images.githubusercontent.com/3778674/201758239-9a45c1b3-67e4-40f7-a372-e5b9a5d250ca.png)

![image](https://user-images.githubusercontent.com/3778674/201758369-2014175e-4cfa-44a3-8137-c68c41e71eea.png)

![image](https://user-images.githubusercontent.com/3778674/201758598-5b3f6b32-330a-4a72-8b6e-cc5665e18b62.png)

**Comment by paolomatarazzo** [comment]
@CoolDadTx as @jirenugo guessed, you are missing the LsaCfgFlags under HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Lsa . Please let us know if that fixes the problem

**Comment by paolomatarazzo** [comment]
Closing this issue, @CoolDadTx let us know if you still have issues

**Comment by CoolDadTx** [comment]
@paolomatarazzo  I have not been able to verify the fix yet. However I still believe the docs need work. The entry you're asking me to add isn't configured by default on any Win10/11 machine I've checked. So even without doing anything on my machine, the latest Windows Update breaks stored Windows credentials. I don't see how anyone without inside knowledge of how this works would be able to figure out the steps they need to follow to get it resolved. The particular registry entry isn't listed in a section that most people would probably related to a default install of Windows. Hence the whole purpose of my reporting of this issue. There needs to be clear steps to follow for a vanilla copy of Windows to identify and resolve this issue.

**Comment by CoolDadTx** [comment]
@paolomatarazzo @jirenugo the registry entry did not work. When I use `mstsc` to connect to a VM running on the same machine (as I always have been before the Windows update) then it rejects the saved credentials and requires me to enter them again. 

![image](https://user-images.githubusercontent.com/3778674/202075927-cd5893f4-42e6-480e-acfa-d4a8a3202851.png)

**Comment by paolomatarazzo** [comment]
@CoolDadTx thank you for the feedback, we will update the documentation as need to make it clearer. @jirenugo to take a closer look

**Comment by CoolDadTx** [comment]
@paolomatarazzo @jirenugo FYI the behavior is inconsistent. Perhaps the issue is just with mstsc. The machine I'm trying to connect to is a Hyper-V VM. I have a PS script that starts Hyper-V and the VM if they are not already running and connects via the standard Hyper-V client. The credentials work here and I can get into the VM. However my script then connects using mstsc and that is when it says the credentials are invalid. So the Hyper-V client seems to be able to read credentials but mstsc cannot. In all cases the account is a local account on the VM.

**Comment by paolomatarazzo** [comment]
@CoolDadTx I would suggest opening a support case to troubleshoot your specific issue. I don't think this is the best place to troubleshoot.

**Comment by mariusgreuel** [comment]
@CoolDadTx Setting LsaCfgFlags to zero did not work for me either, System Information showed Credential Guard still enabled. I do not have an UEFI lock enabled.

The chapter 'Disabling Virtualization-Based Security' mentions `EnableVirtualizationBasedSecurity` and `RequirePlatformSecurityFeatures`. Setting both to zero (plus reboot) actually does turn Credential Guard off, while all remaining VBS features appear still running according to System Information.

To sum up:

```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Lsa]
"LsaCfgFlags"=dword:00000000

[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\DeviceGuard]
"EnableVirtualizationBasedSecurity"=dword:00000000
"RequirePlatformSecurityFeatures"=dword:00000000
```

The issue here is really that 22H2/Credential Guard broke saving passwords for remote desktop.

**Comment by paolomatarazzo** [comment]
@jirenugo can you take a closer look at this issue?

**Comment by jirenugo** [comment]
@mariusgreuel did you reboot after setting LsaCfgFlags to 0? Can you share a snapshot of the reg keys and the event logs from a repro of credential guard still running with LsaCfgFlags set to 0?

**Comment by paolomatarazzo** [comment]
closing due to a lack of response. @mariusgreuel  please reopen the case if you still have the issue

**Comment by Fetakungen** [comment]
Hi, i'm reopening this as or threading it as this is still presist, i've seen it at 4 diffrent sites which uses Remote Apps, The problem i've found is the same as above when not using Credential Guard, But using gpo to disable device guard fixes Saved Cred issue.

BUT, as "my" sites uses credguard and delegation it's not the main issue, Users Auth just fine with passthrough but when logged on the a RDP connections to other servers wont work and refers to no connection to DC, this happens on Windows 1122h2 and is not apperent on h1.

![122](https://user-images.githubusercontent.com/56397651/212038648-430eb948-c1c5-4cc8-9932-d8151da21623.jpg)

On the other hand disableing Cred Guard and entering Credentials manually, solved the issue on the Remote machine and you can access network resources. I have the issue as said on 4 different sites (domains) and can easily replicate the issue this occurs for both full rdp or remote app.

**Comment by paolomatarazzo** [comment]
Hello @Fetakungen the issue you describe may be related to Kerberos unconstrained delegation .For more information, check here: https://learn.microsoft.com/windows/security/identity-protection/credential-guard/credential-guard-considerations#kerberos-considerations.
If the issue persist, I encourage you to open a support ticket

**Comment by EdwardJDill** [comment]
@Fetakungen - I have been having exactly the same issue you described.  Have you found a solution yet?

**Comment by CokeZero21** [comment]
@Fetakungen  @EdwardJDill 

Same exact pattern with all my remoteapp

If i retyped  my password, it connects

![image](https://user-images.githubusercontent.com/95649597/221222024-9432941e-cd86-4a2e-9184-1c46d448f616.png)

**Comment by carehart** [comment]
I am (and have been) in the same boat as @CoolDadTx. Same values (or lack of them) in registry, in gpedit, in event viewer.

Did you ever resolve things, @CoolDadTx?

FWIW, I will add that it's only one machine that I seem to be unable to re-use saved RDP passwords (to any machine). This has been for months, over many restarts. I've even gone into credential manager (on the client doing the RDP) and cleared the credentials of the server I connect to most (a suggestion recommended elsewhere), and that did not help.

It is indeed frustrating having to enter the password over and over, and all the more when the machine I'm connecting to is just another in my own network environment (so that some of the advanced security enabled for RDP is less important than if I was connecting over the internet--at least I'd think, someone may school me otherwise on that point).

I hope that along with others also chiming in here we may be able to get things resolved. Like he explained, all the resources I've ever found on this fail to resolve this ongoing problem.

**Comment by CoolDadTx** [comment]
@carehart I switched to Devolver's Remote Desktop Manager.

**Comment by carehart** [comment]
Yep, switching to an alternative RDP client is indeed another way to **avoid** the problem. :-) I hope someone may still have motivation to pursue solving it, but I realize there may not be such motivation among enough necessary people to make it happen.
